contracts/RoyaltyFeeSetterV2.sol

```diff
@@ -0,0 +1,191 @@
+            _feeInfoParts
+        );
+    }
+
+    /**
+     * @notice Update royalty info for collection
+     * @dev Only to be called if there msg.sender is the setter
+     * @param _collection address of the NFT contract
+     * @param _setter address that sets the receiver
+     * @param _feeInfoParts fee info parts
+     */
+    function updateRoyaltyInfoPartsForCollectionIfSetter(
+        address _collection,
+        address _setter,
+        RoyaltyFeeTypes.FeeInfoPart[] memory _feeInfoParts
+    ) external {
+        address currentSetter = royaltyFeeRegistryV2
+            .royaltyFeeInfoPartsCollectionSetter(_collection);
+        if (msg.sender != currentSetter) {
+            revert RoyaltyFeeSetterV2__NotCollectionSetter();
+        }
+        royaltyFeeRegistryV2.updateRoyaltyInfoPartsForCollection(
+            _collection,
+            _setter,
+            _feeInfoParts
+        );
+    }
+
+    /**
+     * @notice Update royalty info for collection
+     * @dev Can only be called by contract owner (of this)
+     * @param _collection address of the NFT contract
+     * @param _setter address that sets the receiver
+     * @param _feeInfoParts fee info parts
+     */
+    function updateRoyaltyInfoPartsForCollection(
+        address _collection,
+        address _setter,
+        RoyaltyFeeTypes.FeeInfoPart[] memory _feeInfoParts
+    ) external onlyOwner {
+        royaltyFeeRegistryV2.updateRoyaltyInfoPartsForCollection(
+            _collection,
+            _setter,
+            _feeInfoParts
+        );
+    }
+
+    /**
+     * @notice Update owner of royalty fee registry
+     * @dev Can be used for migration of this royalty fee setter contract
+     * @param _owner new owner address
+     */
+    function updateOwnerOfRoyaltyFeeRegistryV2(address _owner)
+        external
+        onlyOwner
+    {
+        IOwnable(address(royaltyFeeRegistryV2)).transferOwnership(_owner);
+    }
+
+    /**
+     * @notice Update royalty info for collection
+     * @param _royaltyFeeLimit new royalty fee limit (500 = 5%, 1,000 = 10%)
+     */
+    function updateRoyaltyFeeLimit(uint256 _royaltyFeeLimit)
+        external
+        onlyOwner
+    {
+        royaltyFeeRegistryV2.updateRoyaltyFeeLimit(_royaltyFeeLimit);
+    }
+
+    /**
+     * @notice Update information and perform checks before updating royalty fee registry
+     * @param _collection address of the NFT contract
+     * @param _setter address that sets the receiver
+     * @param _feeInfoParts fee info parts
+     */
+    function _updateRoyaltyInfoPartsForCollectionIfOwnerOrAdmin(
+        address _collection,
+        address _setter,
+        RoyaltyFeeTypes.FeeInfoPart[] memory _feeInfoParts
+    ) internal {
+        address currentSetter = royaltyFeeRegistryV2
+            .royaltyFeeInfoPartsCollectionSetter(_collection);
+        if (currentSetter != address(0)) {
+            revert RoyaltyFeeSetterV2__SetterAlreadySet();
+        }
+        if (
```

Comment:
I'm curious about this check, I see that it's also in the V1. What's the purpose of it, and what if the contract doesn't implement `supportsInterface` ?

Reply:
If the contract doesn't support `supportsInterface` then it's either a:
- any non-NFT contract that doesn't support EIP165
- badly implemented NFTs

If we don't check this, then any contract could be added like an ERC20.
Sure a non-NFT contract could return true to `supportsInterface(any id)`, but this is a safety check: it doesn't assert that the contract is well implemented, but it's better than nothing.